pandas/core/frame.py

```diff
@@ -393,7 +394,13 @@ def _constructor_expanddim(self):
     # ----------------------------------------------------------------------
     # Constructors
 
-    def __init__(self, data=None, index=None, columns=None, dtype=None, copy=False):
+    def __init__(self,
+                 data: Optional[Union[np.ndarray, dict,
```

Comment:
Need to put `Iterable` in there but it starts erroring if I do that (same for both `abc.Iterable` and `typing.Iterable`):

> pandas/core/frame.py:485: error: Argument 1 to "len" has incompatible type "Iterable[Union[Any, Iterable[str]]]"; expected "Sized"

Also need to provide key, values for `dict` according to docstring:

> Dict can contain Series, arrays, constants, or list-like objects

@WillAyd or just `dict` is ok?

Reply:
Use `Dict` from typing

Reply:
Just `Dict` or need to specify types for key, values inside it too?
Any idea why the error with `Iterable`